src/metabase/driver/postgres.clj

```diff
@@ -770,3 +773,49 @@
   ;; This could be incorrect if Postgres has been compiled with a value for NAMEDATALEN other than the default (64), but
   ;; that seems unlikely and there's not an easy way to find out.
   63)
+
+(defn- format-copy
+  [_clause table]
+  [(str "COPY " (sql/format-entity table))])
+
+(sql/register-clause! ::copy format-copy :insert-into)
+
+(defn- format-from-stdin
+  [_clause delimiter]
+  [(str "FROM STDIN NULL " delimiter)])
+
+(sql/register-clause! ::from-stdin format-from-stdin :from)
+
+(defn- db->connection
+  [db-or-id]
+  (.unwrap (jdbc/get-connection (sql-jdbc.conn/db->pooled-connection-spec db-or-id))
+           PgConnection))
+
+(defn- sanitize-value
+  ;; Per https://www.postgresql.org/docs/current/sql-copy.html#id-1.9.3.55.9.2
```

Comment:
Might save some reading time if we include the relevant quote from that section (needs formatting though)
```suggestion
  ;; Per https://www.postgresql.org/docs/current/sql-copy.html#id-1.9.3.55.9.2
  ;; "Backslash characters (\) can be used in the COPY data to quote data characters that might otherwise be taken as row or column delimiters. In particular, the following characters must be preceded by a backslash if they appear as part of a column value: backslash itself, newline, carriage return, and the current delimiter character."
```